cmd/thriftbreak/main.go

```diff
@@ -21,23 +21,54 @@
 package main
 
 import (
+	"encoding/json"
+	"errors"
 	"flag"
 	"fmt"
+	"io"
 	"log"
 	"os"
 
+	"go.uber.org/thriftrw/internal/compare"
 	"go.uber.org/thriftrw/internal/git"
 )
 
 func main() {
-	if err := run(os.Args[1:]); err != nil && err != flag.ErrHelp {
+	if err := run(os.Args[1:]); err != nil && !errors.Is(err, flag.ErrHelp) {
 		log.Fatalf("%+v", err)
 	}
 }
 
+func readableOutput(w io.Writer) func(compare.Diagnostic) error {
+	return func(diagnostic compare.Diagnostic) error {
+		// fmt.Println(diagnostic.String())
+		if _, err := fmt.Fprintln(w, diagnostic.String()); err != nil {
+			return fmt.Errorf("failed to output a lint error: %v", err)
+		}
+
+		return nil
+	}
+}
+
+func jsonOutput(w io.Writer) func(compare.Diagnostic) error {
+	enc := json.NewEncoder(w)
+
+	return func(diagnostic compare.Diagnostic) error {
+		// Encode adds a trailing newline.
+		if err := enc.Encode(diagnostic); err != nil {
```

Comment:
Do we care about failure cases being tested here?

Reply:
yeah, im not sure how to make serialization of a struct that we own fail. fields are strings too so i couldnt figure out how to make a test for catching an error.